### ISSUE TITLE
Pre/Post-tasks fail on windows

### ISSUE THREAD

**Issue by jungerm2** [fabric/fabric#2202]
On windows 10 (with Python 3.9.2, Fabric 2.6.0, Paramiko 2.9.2, Invoke 1.6.0), the below example doesn't run as expected:

```python
from fabric import task

@task
def e1(c):
    c.run("echo 1")


@task(pre=[e1])
def e2(c):
    c.run("echo 2")
```

Specifically it leads to the following error: 
```
Traceback (most recent call last):
  File "c:\users\sjung\miniconda3\envs\mpldev\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\sjung\miniconda3\envs\mpldev\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\sjung\miniconda3\envs\mpldev\Scripts\fab.exe\__main__.py", line 7, in <module>
  File "c:\users\sjung\miniconda3\envs\mpldev\lib\site-packages\invoke\program.py", line 384, in run
    self.execute()
  File "c:\users\sjung\miniconda3\envs\mpldev\lib\site-packages\invoke\program.py", line 566, in execute
    executor.execute(*self.tasks)
  File "c:\users\sjung\miniconda3\envs\mpldev\lib\site-packages\invoke\executor.py", line 129, in execute
    result = call.task(*args, **call.kwargs)
  File "c:\users\sjung\miniconda3\envs\mpldev\lib\site-packages\invoke\tasks.py", line 127, in __call__
    result = self.body(*args, **kwargs)
  File "C:\Users\sjung\Documents\Projects\WebServer\fabfile.py", line 135, in e1
    c.run("echo 1")
  File "c:\users\sjung\miniconda3\envs\mpldev\lib\site-packages\invoke\context.py", line 95, in run
    return self._run(runner, command, **kwargs)
  File "c:\users\sjung\miniconda3\envs\mpldev\lib\site-packages\invoke\context.py", line 102, in _run
    return runner.run(command, **kwargs)
  File "c:\users\sjung\miniconda3\envs\mpldev\lib\site-packages\invoke\runners.py", line 380, in run
    return self._run_body(command, **kwargs)
  File "c:\users\sjung\miniconda3\envs\mpldev\lib\site-packages\invoke\runners.py", line 431, in _run_body
    self.start(command, self.opts["shell"], self.env)
  File "c:\users\sjung\miniconda3\envs\mpldev\lib\site-packages\invoke\runners.py", line 1291, in start
    self.process = Popen(
  File "c:\users\sjung\miniconda3\envs\mpldev\lib\subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "c:\users\sjung\miniconda3\envs\mpldev\lib\subprocess.py", line 1420, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
OSError: [WinError 87] The parameter is incorrect
```

A similar error occurs when trying a post-task:

```python
from fabric import task

@task
def e2(c):
    c.run("echo 2")

@task(post=[e2])
def e1(c):
    c.run("echo 1")
```

I found several [related errors](https://github.com/fabric/fabric/issues?q=WinError+87) but they seem to all be caused due to parallel tasks. However here we are explicitly setting pre/post dependencies so nothing should run in parallel. Any ideas?

**Comment by davidjmemmett** [comment]
Hi @jungerm2,

I believe this issue is fixed in the latest Fabric release (2.7.0) - can you please retry?

Cheers,
David

**Comment by jungerm2** [comment]
Sorry, for the delay. I've just tested this with the latest version of fabric and it seems to indeed have been resolved. Thanks!
